hollow/src/test/java/com/netflix/hollow/api/producer/HollowProducerBlobStorageCleanerTest.java

```diff
@@ -0,0 +1,122 @@
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.hollow.api.producer;
+
+import com.netflix.hollow.api.producer.HollowProducer.Populator;
+import com.netflix.hollow.api.producer.HollowProducer.WriteState;
+import com.netflix.hollow.api.producer.fs.HollowFilesystemBlobStorageCleaner;
+import com.netflix.hollow.api.producer.fs.HollowFilesystemPublisher;
+import com.netflix.hollow.core.write.objectmapper.HollowPrimaryKey;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class HollowProducerBlobStorageCleanerTest {
+
+    private static final String SCRATCH_DIR = System.getProperty("java.io.tmpdir");
+
+    private File publishDir;
+
+    @Before
+    public void setUp() {
+        publishDir = new File(SCRATCH_DIR, "publish-dir");
+        publishDir.mkdir();
+    }
+
+    @Test
+    public void cleanSnapshotsWithDefaultValue() {
+        HollowProducer.Publisher publisher = new HollowFilesystemPublisher(publishDir);
+        HollowProducer.BlobStorageCleaner blobStorageCleaner = new HollowFilesystemBlobStorageCleaner(publishDir);
+        HollowProducer producer = HollowProducer.withPublisher(publisher)
+                                                .withBlobStorageCleaner(blobStorageCleaner)
+                                                .build();
+
+        /// initialize the data -- classic producer creates the first state in the delta chain. 
+        producer.runCycle(new Populator() {
+            public void populate(WriteState state) throws Exception {
+                state.add(new TypeA(1, "one", 1));
+            }
+        });
+
+        HollowIncrementalProducer incrementalProducer = new HollowIncrementalProducer(producer);
+        incrementalProducer.addOrModify(new TypeA(2, "two", 100));
+        incrementalProducer.runCycle();
+        incrementalProducer.addOrModify(new TypeA(3, "three", 1000));
+        incrementalProducer.runCycle();
+        incrementalProducer.addOrModify(new TypeA(4, "two", 100));
+        incrementalProducer.runCycle();
+        incrementalProducer.addOrModify(new TypeA(5, "three", 1000));
+        incrementalProducer.runCycle();
+
+        File[] files = listFiles(HollowProducer.Blob.Type.SNAPSHOT.prefix);
+        List<String> fileNames = getFileNames(files);
+        assert files.length == 5;
```

Comment:
The unit test should use Junit Assert. Can you switch them?

Reply:
done, thanks for point this out